### PR TITLE
Starlight colors with parallax, small parallax code clean-up

### PR DESCRIPTION
**Pull request by Time-Green**
The space gas parallax now colors starlight in the color of the gas: 

image

<details>
  <summary>Others</summary>
  
image
image
image

</details>

I also cleaned up parallax random code a bit, making it easier to add new parallaxes and add minor effects to them

The radioactive nebula parallax thing has also been moved to SSparallax to make it easier and a bit more sane to change

:cl:
add: Starlight will color with space gas parallax
code: Cleans up random parallax code / radioactive nebula parallax code
/:cl:

## Why it's good for the game

We can change the starlight color now (mostly thanks to @LemonInTheDark making light code easier to work with), so I thought it'd be fun to actually do! If people hate it, I'll turn it off and we can just call this a code clean-up PR

### CODE REVIEW

**Comment by AdipemDragon** [comment]
finally! I've been waiting for this literally ever since the first pr that gave space light!

File: code/controllers/subsystem/processing/station.dm [hunk 1]
```diff
@@ -21,6 +21,7 @@ PROCESSING_SUBSYSTEM_DEF(station)
 	#endif
 
 	announcer = new announcer() //Initialize the station's announcer datum
+	SSparallax.random_layer.apply_global_effects() //Apply station effects that parallax might have
```
Comment (Time-Green) [inline code/controllers/subsystem/processing/station.dm:24]: This seemed the most sensical place to put this? SSparallax inits before globals do, so we can't really change anything there. SSstation is already used for stuff like this with station traits (and I'm familiar with it)

**Comment by Dorsisdwarf** [comment]
god bless the pisslight shall be real

**Comment by Wallemations** [comment]
We in a Corsair k55 RGB gaming keyboard now

File: code/controllers/subsystem/parallax.dm [hunk 2]
```diff
@@ -57,3 +65,32 @@ SUBSYSTEM_DEF(parallax)
 		if (MC_TICK_CHECK)
 			return
 	currentrun = null
+
+/// Generate a random layer for parallax
+/datum/controller/subsystem/parallax/proc/set_random_parallax_layer(picked_parallax)
+	if(picked_parallax == PARALLAX_NONE)
+		return
+
+	random_layer = new picked_parallax(null, null, TRUE) //rip no named params, but we set template to TRUE so we don't del without a hud owner
```
Comment (Jacquerel) [inline code/controllers/subsystem/parallax.dm:74]:
```suggestion
	random_layer = new picked_parallax(null, null, /* template = */ TRUE) //rip no named params, but we set template to TRUE so we don't del without a hud owner
```
Mothblocks has suggested this pattern before where named args are not possible, would be nice to add all 3

**Review by Jacquerel — CHANGES_REQUESTED** [review]
I want to see if this makes it easier to make space look cool during aurora caelis (or if it's too slow) 
Not directly related to this pr though

**Comment by Time-Green** [comment]
@Jacquerel I've added the inline comments for the params!

**Comment by LemonInTheDark** [comment]
PISSLIGHT PISSLIGHT (fuck you the blue was soo pretty, that green's alright tho. Might be worth making starlight a bit less vibrant then the parallax color, could use that generically with HSV I think)

**Comment by Time-Green** [comment]
> PISSLIGHT PISSLIGHT (fuck you the blue was soo pretty, that green's alright tho. Might be worth making starlight a bit less vibrant then the parallax color, could use that generically with HSV I think)

Default starlight still happens when there's no gas-clouds, but yeah maybe the saturation of the space-light could be turned down for the colored gas-clouds

**Comment by Time-Green** [comment]
@Jacquerel we could probably do RGB parallax / aurora caelis by coloring the gas cloud and punting the appropriate light settings into a background subsystem

**Pull request merged**